kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkAdminServiceImpl.java

```diff
@@ -1228,6 +1230,17 @@ public void setFirewallNatConfiguration(List<FirewallNatConfig> natConfigs) thro
         return wifiHotspotInfoList;
     }
 
+    @Override
+    public List<WifiChannel> getWifiFrequencies(String ifaceName) throws KuraException {
+        List<WifiChannel> frequencies = IwListChannelTool.probeChannels(ifaceName, this.executorService);
+        return frequencies;
```

Comment:
maybe this can be simply: `return IwListChannelTool.probeChannels(ifaceName, this.executorService);`

Reply:
OK.